### PR TITLE
mathematica: 13.1.0 -> 13.2.0

### PR DESCRIPTION
**Pull request by herberteuler**
###### Things done

This PR upgrades Mathematica to 13.2.0.
Since this version, both installations of local documentation and no local documentation are supported, with the local documentation installation being the default. (#207087)

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/math/mathematica/versions.nix [hunk 1]
```diff
@@ -1,6 +1,22 @@
...
+    installer = "Mathematica_13.2.0_LINUX.sh";
+    webdoc = true;
+  }
+  {
+    version = "13.2.0";
+    lang = "en";
+    language = "English";
+    sha256 = "sha256-YRUvl2H9SwpwDZx04ugd7ZnK5G+t88bzAObXsGGVhk0=";
+    installer = "Mathematica_13.2.0_BNDL_LINUX.sh";
+    webdoc = false;
```
Comment (doronbehar) [inline pkgs/applications/science/math/mathematica/versions.nix:18]: Maybe we can parse the `installer` and figure out whether documentation is included or not?

Comment (herberteuler) [reply pkgs/applications/science/math/mathematica/versions.nix]: Updated.

**Comment by doronbehar** [comment]
I tried to add my installer to `versions.nix`, without success yet. Will continue to debug this tomorrow. Other then that, changes look good to me.

**Comment by bjornfor** [comment]
Ready to merge?

**Comment by doronbehar** [comment]
Sorry for ghosting, it was a busy week. Can you add perhaps a comment, somewhere at `versions.nix`, specifying what Nix command should one run in order to add their installer to the list? I don't get along with all of these `nix store` commands.. See https://github.com/NixOS/nix/issues/6210

**Comment by doronbehar** [comment]
Adding the installer to the list is explained in the `requireFile`'s message, but how to get it's hash isn't, and I think that's where I fail. Here's what I ran:

```sh
nix-store --query --hash $(nix-store --add-fixed sha256 Mathematica_13.1.0_LINUX.sh)
```

**Comment by herberteuler** [comment]
> Sorry for ghosting, it was a busy week. Can you add perhaps a comment, somewhere at `versions.nix`, specifying what Nix command should one run in order to add their installer to the list? I don't get along with all of these `nix store` commands.. See [NixOS/nix#6210](https://github.com/NixOS/nix/issues/6210)

I've just added a comment, please check it out.

File: pkgs/applications/science/math/mathematica/versions.nix [hunk 2]
```diff
@@ -1,6 +1,26 @@
 { lib, requireFile }:
 
+/*
+ * To calculate the hash of an installer, use a command like this:
+ *
+ *   nix --extra-experimental-features nix-command hash file <installer-file>
```
Comment (doronbehar) [inline pkgs/applications/science/math/mathematica/versions.nix:6]: That worked! Thanks.

**Comment by doronbehar** [comment]
I added my `13.1.0` webdoc=false version I have on my computer, using the `nix hash file` command specified. This PR is ready IMO, can merge when CI is green.

**Comment by herberteuler** [comment]
> I added my `13.1.0` webdoc=true version I have on my computer, using the `nix hash file` command specified. This PR is ready IMO, can merge when CI is green.

Oh right, I recalled the command from my bash history, when `nix hash` was not available. Now we can use the simpler form instead.

**Pull request merged**